add a NetDef argument to override the file path for tracing results

duc0 opened the pull request:
Summary:
Add a NetDef argument to override the file path for tracing results
The argument is called tracing_filepath_override. If it is not specified, default to what is specified in the command line flag caffe2_net_async_tracing_filepath.

Differential Revision: D8680490

bddppq commented on caffe2/utils/proto_utils.h:
```diff
@@ -291,6 +291,10 @@ class ArgumentHelper {
 // **** Arguments Utils *****
 
+// Helper methods to check if an argument from OperatorDef or NetDef exists.
+bool HasArgument(const OperatorDef& def, const string& name);
```
There is already `ArgumentHelper::HasArgument`

duc0 committed:
add a NetDef argument to override the file path for tracing results (#8998)

Summary:
Closes https://github.com/pytorch/pytorch/pull/8998

Add a NetDef argument to override the file path for tracing results
The argument is called tracing_filepath_override. If it is not specified, default to what is specified in the command line flag caffe2_net_async_tracing_filepath.

Differential Revision: D8680490

fbshipit-source-id: 80e2e66195e052bdb38bf40474986bdb45957395
